### PR TITLE
[FIX] web_editor: validate qweb fields upon submission

### PR DESCRIPTION
**Pull request by lade-odoo**
When editing database fields via the web editor, their value are not checked.
Thus, stack traces can come up to the front-end user.

Step to reproduce the issue:
1) Install the E-Learning module and connect to the website
2) On the main website (not the backend), go to Courses > Edit
3) Edit the Next Rank treshold with any non integer string (e.g.: coucou)
A stracktrace will be shown upon save.

Solution: The issue is that there are no validation on the submitted fields,
this can cause stacktraces. A try-catch was used around the parsing of the
input value to catch and properly raise these exceptions to show clean errors.
On top of this, integers were not properly parsed, they were not taking into
account the `thousands_sep` of the user lang (e.g.: 35,000 for 35000).

opw-2819392

### CODE REVIEW

**Review by nle-odoo — APPROVED** [review]
doing some test with integer and float (like having "coucou" or "14,00" as integer, and "coucou" as float) I only see ValueError happening at the line:

```py
value = converter.from_html(Model, Model._fields[field], el)
```
so it would probably be better to handle only that in try/catch (with the same invalid value message), unless I'm missing other use case.

besides that, looks good to me

File: addons/web_editor/models/ir_ui_view.py [hunk 1]
```diff
@@ -6,7 +6,8 @@
 from lxml import etree, html
 
 from odoo.exceptions import AccessError
-from odoo import api, fields, models
+from odoo import api, models, _
+from odoo.exceptions import ValidationError
```
Comment (nle-odoo) [inline addons/web_editor/models/ir_ui_view.py:10]: here it's slightly better to have both import at same like (`import AccessError, ValidationError`) or possibly you could do that and fix the order putting `from odoo` above `from odoo.exceptions`)

**Review by rdeodoo — CHANGES_REQUESTED** [review]
Sorry for the late reply.
I have noticed that in 15.0 it was fixed with a specific fix for float and monetary field with https://github.com/odoo/odoo/pull/85535/files
Why 15 and why just a specific fix for float and monetary, I have no idea and I didn't check further, but your fix seems good. It should just take care of the mentioned PR in the forward port (remove the specific code and keep this generic one?).

If you could also add that PR's commit in your commit message, would be great, thanks

File: addons/web_editor/models/ir_ui_view.py [hunk 2]
```diff
@@ -49,7 +49,11 @@ def save_embedded_field(self, el):
 
         model = 'ir.qweb.field.' + el.get('data-oe-type')
         converter = self.env[model] if model in self.env else self.env['ir.qweb.field']
-        value = converter.from_html(Model, Model._fields[field], el)
+
+        try:
+            value = converter.from_html(Model, Model._fields[field], el)
+        except ValueError:
+            raise ValidationError(_("Invalid value for %s.%s: %s") % (Model._name, field, el.text_content().strip()))
```
[the remark this answers is not in the document]
Comment (lade-odoo) [reply addons/web_editor/models/ir_ui_view.py]: Was on holiday, sorry for the late reaction.
Term was already present in the .pot nonetheless, I adapted the error message (and added it into the .pot).

**Comment by lade-odoo** [comment]
@rdeodoo Are we good to R+ here ? I have noted the changes that needs to be done in 15.0 during the forward port, don't worry !

**Review by rdeodoo — APPROVED** [review]
Welcome back, yep this is LGTM, thanks for the changes

Worth noting: The code of the `from_html` is greatly inspired by the one from `ir.qweb.field.float` a few lines below.

@robodoo r+

**Pull request closed without merging**